.hlint-universum.yaml

```diff
@@ -0,0 +1,414 @@
+- warn: { lhs: sortOn (Down . fst), rhs: Universum.sortWith (Down . fst),
+    note: "'sortWith' will be faster here because it doesn't do caching" }
+- warn: { lhs: sortOn (Down . snd), rhs: Universum.sortWith (Down . snd),
+    note: "'sortWith' will be faster here because it doesn't do caching" }
+
+- warn: { lhs: fmap concat (mapM f s), rhs: Universum.concatMapM f s }
+- warn: { lhs: concat <$> mapM f s, rhs: Universum.concatMapM f s }
+
+# Removed for now since we don't want to make people use (some of) our ad-hoc stuff.
+# - warn: {lhs: fmap concat (forM f s), rhs: Universum.concatForM s f}
+# - warn: {lhs: fmap concat (for f s), rhs: Universum.concatForM s f}
+# - warn: {lhs: concat <$> forM f s, rhs: Universum.concatForM s f}
+# - warn: {lhs: concat <$> for f s, rhs: Universum.concatForM s f}
+
+- suggest:
+    lhs: fmap and (sequence s)
+    rhs: Universum.andM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: and <$> sequence s
+    rhs: Universum.andM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap or (sequence s)
+    rhs: Universum.orM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: or <$> sequence s
+    rhs: Universum.orM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap and (mapM f s)
+    rhs: Universum.allM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: and <$> mapM f s
+    rhs: Universum.allM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap or (mapM f s)
+    rhs: Universum.anyM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: or <$> mapM f s
+    rhs: Universum.anyM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+
+# Unfortunately, these are often bad because they remove a variable name
+# (which usually clarifies things):
+# - suggest: { lhs: (do x <- m; when x a), rhs: Universum.whenM m a }
+# - suggest: { lhs: (do x <- m; unless x a), rhs: Universum.unlessM m a }
+
+- warn: { lhs: whenM (not <$> x), rhs: unlessM x }
+- warn: { lhs: unlessM (not <$> x), rhs: whenM x }
+
+# Oh boy, we sure have many ways of spelling “pure ()”. Also I checked and
+# HLint isn't smart enough to see reordered case branches.
```

Comment:
Rules in `master` have a comment with this link https://github.com/ndmitchell/hlint/issues/434